### PR TITLE
Automaticaly deprecate entire NPM packages during status checks

### PR DESCRIPTION
**Pull request by johnbintz-tidelift**
If every version of an NPM package is deprecated, the entire package is
deprecated. If there's a new release with a new, non-deprecated version,
it will become undeprecated.

Thanks taking the time to contribute. This template should help guide you through the process of creating a pull request for review. Please erase any part of this template that is not relevant to your pull request:

- [X] Have you followed the guidelines for [contributors](http://docs.libraries.io/contributorshandbook)?
- [X] Have you checked to ensure there aren't other open pull requests on the repository for a similar change?
- [X] Is there a corresponding ticket for your pull request?
- [X] Have you written new tests for your changes?
- [X] Have you successfully run the project with your changes locally?

If so then please replace this section with a link to the ticket(s) it addressed, an explanation of your change and why you think we should include it. Thanks again!

### CODE REVIEW

File: app/models/package_manager/base.rb [hunk 1]
```diff
@@ -227,6 +228,10 @@ def self.project_find_names(project_name)
       [project_name]
     end
 
+    def self.entire_package_deprecation_info(name)
```
Comment (tiegz) [inline app/models/package_manager/base.rb]:
```suggestion
    def self.entire_package_deprecation_info(_name)
```

Comment (katzj) [inline app/models/package_manager/base.rb]: This is a bikeshed but I'm not loving this class method name. `entire_package_` feels unnecessary? We don't do most things at a version level and this method is on a project so by definition it's about the project.

So maybe this is just `self.deprecation_info`? Are we planning on using the `message` field?

Comment (johnbintz-tidelift) [reply app/models/package_manager/base.rb]: I'm fine with renaming it. We will use the `message` field in the next piece of deprecation information being passed on, which will come in a week or so.

Comment (johnbintz-tidelift) [reply app/models/package_manager/base.rb]: I'm fine with renaming it. We will be using the `message` field for work coming in a week or two.

**Pull request merged**